### PR TITLE
Add migration(Reservedip, nsg, route) cmdlets

### PR DESCRIPTION
**Pull request by DeepakRajendranMsft**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@DeepakRajendranMsft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by DeepakRajendranMsft** [comment]
This failure is not related to my changes, the previous commit passed all tests. Could you please trigger the build again.

d:\workspace\powershell\src\Common\Commands.Common.Authentication\Models\DiskDataStore.cs(42): error : Microsoft.WindowsAzure.Commands.Test.Websites.SaveAzureWebsiteLogTests.SaveAzureWebsiteLogWithNoFileExtensionTest [FAIL]

**Comment by markcowl** [comment]
@DeepakRajendranMsft Yep, rerunning

File: src/ServiceManagement/Common/Commands.ScenarioTest/ServiceManagement/ScenarioTests.cs [hunk 1]
```diff
@@ -182,6 +182,33 @@ public void TestMigrateAzureVNet()
...
         [Trait(Category.AcceptanceType, Category.BVT)]
+        public void TestMigrationNetworkSecurityGroup()
+        {
+            this.RunPowerShellTest("Test-MigrationNetworkSecurityGroup");
+        }
+
+        [Fact]
+        [Trait(Category.Service, Category.ServiceManagement)]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        [Trait(Category.AcceptanceType, Category.BVT)]
```
Comment (markcowl) [inline src/ServiceManagement/Common/Commands.ScenarioTest/ServiceManagement/ScenarioTests.cs]: Note that the other AcceptanceType attribute is redundant.  CheckIn tests are automatically included as BVTs

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MigrationValidateContext.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                    result.ValidationMessages[i] = new ValidationMessage
+                    {
+                        ResourceName = validationResponse.ValidationMessages[i].ResourceName,
+                        ResourceType = validationResponse.ValidationMessages[i].ResourceType,
+                        Category = validationResponse.ValidationMessages[i].Category,
+                        Message = validationResponse.ValidationMessages[i].Message,
+                        VirtualMachineName = validationResponse.ValidationMessages[i].VirtualMachineName
+                    };
+                }
+                result.Result = "Validation failed.  Please see ValidationMessages for details";
```
Comment (johnmic) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MigrationValidateContext.cs]: There can be informational/success messages. Existence of a message does not mean validation failed.

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+            if (this.Validate.IsPresent)
+            {
+                ExecuteClientActionNewSM(
+                null,
+                CommandRuntime.ToString(),
+                () => this.NetworkClient.NetworkSecurityGroups.ValidateMigration(this.NetworkSecurityGroupName),
+                (operation, service) =>
+                {
+                    var context = MigrationValidateContextHelper.ConvertToContext(operation, service);
```
Comment (johnmic) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs]: Please write each validation message to the object stream, not a wrapper response object. Do this for all validation APIs.

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+using Microsoft.Azure;
+using Microsoft.WindowsAzure.Management.Network;
+using System.Management.Automation;
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.IaaS.Network
+{
+    /// <summary>
+    /// Migrate ASM Network Security Group to ARM
+    /// </summary>
+    [Cmdlet(VerbsCommon.Move, "AzureNetworkSecurityGroup"), OutputType(typeof(OperationStatusResponse))]
```
Comment (johnmic) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs]: OutputType is different for validation APIs?

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs (hunk 3 above)
Comment (DeepakRajendranMsft) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs]: this seems to be the current experience for all existing migration cmdlets (this also what happens for all azure cmdlets i believe). @huangpf @hyonholee  any comments?

we can discuss this offline without stopping this PR for now

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs (hunk 4 above)
Comment (DeepakRajendranMsft) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs]: outputtype is not same for several apis (mainly get) Dont think we follow a strict validation 

@hyonholee @huangpf Seems to be case for all migration cmdlets. any comments? should we change?

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MigrationValidateContext.cs (hunk 2 above)
Comment (DeepakRajendranMsft) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MigrationValidateContext.cs]: Let me talk offline with @huangpf  and Priyank on this. If we plan to change this, we have to change for all migration cmdlets.

**Comment by markcowl** [comment]
@DeepakRajendranMsft Looks like issues with the Azure.psd1 module and some CredScan issues with the help.  Also, you will need to merge with upstream, as there are merge conflicts.

**Comment by DeepakRajendranMsft** [comment]
@markcowl Thanks!

I didnt change anything in azure.psd1 and with regards to help. i just added some description of cmdlets and parameters. is there a way of identifying is exactly wrong with the help. Also, is this issue with only CI or with other PRs?

**Comment by markcowl** [comment]
@DeepakRajendranMsft You will need to retarget this PR at release-2.0.0

**Comment by DeepakRajendranMsft** [comment]
@markcowl will do

**Comment by DeepakRajendranMsft** [comment]
created a new pull request against release-2.0.0 branch

**Pull request closed without merging**